RELEASE.md

```diff
@@ -0,0 +1,34 @@
+# Release instructions for jupyter-sphinx
+
+Jupyter Sphinx uses a GitHub action to automatically push a new release to
+PyPI when a GitHub release is added.
+
+To cut a new Jupyter Sphinx release, follow these steps:
+
+* Ensure that all tests are passing on master.
+* In [`_version.py`](https://github.com/jupyter/jupyter-sphinx/blob/master/jupyter_sphinx/_version.py),
+  change the "release type" section to "final" e.g.:
+
+  ```python
+  version_info = (0, 2, 3, "final")
+  ```
+* Make a release commit and push to master
+  
+  ```
+  git add jupyter_sphinx/_version.py
+  git commit -m "RLS: 0.2.3"
+  git push upstream master
+  ```
+* [Create a new github release](https://github.com/jupyter/jupyter-sphinx/releases/new).
```

Comment:
Is this the same as making an annotated tag of a commit and pushing the tag? If yes: do you think it would be worth to mention?

Reply:
yep - just pushed an updated explanation